crates/rattler_solve/src/resolvo/conda_util.rs

```diff
@@ -49,23 +49,38 @@ pub(super) fn compare_candidates<'a>(
 
     // Otherwise, compare the dependencies of the variants. If there are similar
     // dependencies select the variant that selects the highest version of the dependency.
-    let a_match_specs = solver
+    let a_dependencies = solver
         .get_or_cache_dependencies(a)
-        .requirements
-        .iter()
-        .map(|id| (*id, pool.resolve_version_set(*id)));
-    let b_match_specs = solver
+        .expect("should not get here, resolution process aborted");
+    let b_dependencies = solver
         .get_or_cache_dependencies(b)
-        .requirements
-        .iter()
-        .map(|id| (*id, pool.resolve_version_set(*id)));
-
-    let b_specs_by_name: HashMap<_, _> = b_match_specs
-        .map(|(spec_id, _)| (pool.resolve_version_set_package_name(spec_id), spec_id))
-        .collect();
-
-    let a_specs_by_name =
-        a_match_specs.map(|(spec_id, _)| (pool.resolve_version_set_package_name(spec_id), spec_id));
+        .expect("should not get here, resolution process aborted");
+
+    // If the MatchSpecs are known use these
+    // map these into a HashMap<PackageName, VersionSetId>
+    // for comparison later
+    let (a_specs_by_name, b_specs_by_name) = if let Dependencies::Known(a_known) = a_dependencies {
```

Comment:
You can simplify this a little:

```suggestion
    let (a_specs_by_name, b_specs_by_name) = if let (Dependencies::Known(a_known), Dependencies::Known(b_known)) = (a_dependencies, b_dependencies) {
```